File: src/main/java/math/map/Map2d.java
package math.map;

import java.util.Arrays;

public class Map2d {
    public Double[] axis;
    public Double[] data;


    public Map2d() {
        super();
    }

    public Map2d(Double[] axis, Double[] data) {
        this.axis = axis;
        this.data = data;
    }

    public Map2d(Map2d map2d) {
        this.axis = Arrays.copyOf(map2d.axis, map2d.axis.length);
        this.data = Arrays.copyOf(map2d.data, map2d.data.length);
    }
}

File: src/main/kotlin/domain/math/map/Map2d.kt
package domain.math.map

class Map2d {
    var axis: Array<Double>
    var data: Array<Double>

    constructor() {
        axis = emptyArray()
        data = emptyArray()
    }

    constructor(axis: Array<Double>, data: Array<Double>) {
        this.axis = axis
        this.data = data
    }

    constructor(map2d: Map2d) {
        this.axis = map2d.axis.copyOf()
        this.data = map2d.data.copyOf()
    }
}
